YOLayout/YOLayout_IMP.m

```diff
@@ -142,24 +149,25 @@ - (CGRect)setFrame:(CGRect)frame inRect:(CGRect)inRect view:(id)view options:(YO
   if ((options & YOLayoutOptionsSizeToFitVertical) == 0) {
     frame.size.height = originalFrame.size.height;
   }
-  
-  CGRect rect = originalFrame;
-  if (!CGRectIsEmpty(inRect)) rect = inRect;
-  
-  if ((options & YOLayoutOptionsAlignCenterVertical) != 0) {
-    frame = YOCGRectToCenterYInRect(frame, originalFrame);
-  }
+
+  // If we passed in 0 for inRect height, then lets set it to the frame height.
+  // This usually happens if we are sizing to fit, and is needed to align below.
+  if (inRect.size.height == 0) inRect.size.height = frame.size.height;
```

Comment:
(Thought i commented here, but not sure what happened to it)

If we're going to shift everything to using `inRect` instead of `rect` and remove the above line that does the CGRectIsEmpty check, we need to do width here as well.

    if (inRect.size.width == 0) inRect.size.width = frame.size.width;

Currently as of 2c93c6c this is breaking horizontal centering (and probably right aligning)

![ios simulator screen shot jan 20 2015 10 44 19 pm](https://cloud.githubusercontent.com/assets/218876/5832197/f18fb498-a0f5-11e4-9765-4475ea5bd458.png)